tests/LDAP/AuthLdap.php

```diff
@@ -1469,6 +1472,112 @@ public function testRemovedUser() {
       $this->boolean((bool)$user->fields['is_deleted'])->isTrue();
    }
 
+   /**
+    * Test restoring users from LDAP
+    *
+    * @extensions ldap
+    *
+    * @return void
+    */
+   public function testRestoredUser() {
+      global $CFG_GLPI;
+
+      $ldap = $this->ldap;
+
+      //put deleted LDAP users in trashbin
+      $CFG_GLPI['user_deleted_ldap'] = 1;
```

Comment:
To prevent most impact on other tests if this one is failing, please restore `$CFG_GLPI` values as soon as you can.

This may affect `$CFG_GLPI` in other tests, if assertion failed:
```php
$CFG_GLPI['x'] = 1;
$result = $user->getFromDB($id);
$this->boolean($result)->isTrue();
...
```
but with this order, `$CFG_GLPI` will be restored to its previous value:
```php
$x_backup = $CFG_GLPI['x'];
$CFG_GLPI['x'] = 1;
$result = $user->getFromDB($id);
$CFG_GLPI['x'] = $x_backup;
$this->boolean($result)->isTrue();
```